### PR TITLE
Enable splitk with convolution

### PR DESCRIPTION
**Pull request by pfultz2**

### CODE REVIEW

File: src/targets/gpu/jit/mlir.cpp [hunk 1]
```diff
@@ -77,7 +77,7 @@ struct mlir_compiler : compiler<mlir_compiler>
         auto* smod = ins->module_inputs().front();
         assert(smod->get_parameter_names().size() == ins->inputs().size() - 1);
         auto gemm_ins = std::find_if(smod->begin(), smod->end(), [&](const auto& i) {
-            return i.name() == "dot" or i.name() == "quant_dot";
+            return contains({"dot", "quant_dot", "convolution", "quant_convolution"}, i.name());
```
Comment (umangyadav) [inline src/targets/gpu/jit/mlir.cpp:80]: Is there a test for this one ? 

If not can you add one please  ? E.g. I'd added one for the GEMMs in this PR https://github.com/ROCm/AMDMIGraphX/pull/3082/files

File: src/targets/gpu/jit/mlir.cpp [hunk 2]
```diff
@@ -77,7 +77,7 @@ struct mlir_compiler : compiler<mlir_compiler>
         auto* smod = ins->module_inputs().front();
         assert(smod->get_parameter_names().size() == ins->inputs().size() - 1);
         auto gemm_ins = std::find_if(smod->begin(), smod->end(), [&](const auto& i) {
```
Comment (umangyadav) [inline src/targets/gpu/jit/mlir.cpp:79]:
```suggestion
        auto gemm_based_ins = std::find_if(smod->begin(), smod->end(), [&](const auto& i) {
```

File: src/targets/gpu/jit/mlir.cpp [hunk 3]
```diff
@@ -77,7 +77,7 @@ struct mlir_compiler : compiler<mlir_compiler>
...
-            return i.name() == "dot" or i.name() == "quant_dot";
+            return contains({"dot", "quant_dot", "convolution", "quant_convolution"}, i.name());
         });
         // check if (a) module is fused (b) contains a dot instruction and (c) perfConfig can not
```
Comment (manupak) [inline src/targets/gpu/jit/mlir.cpp:82]: This comment is outdated now.

File: src/targets/gpu/jit/mlir.cpp [hunk 4]
```diff
@@ -77,7 +77,7 @@ struct mlir_compiler : compiler<mlir_compiler>
...
         // allow fused module
```
[the remark this answers is not in the document]
Comment (umangyadav) [reply src/targets/gpu/jit/mlir.cpp:83]: https://github.com/ROCm/AMDMIGraphX/pull/3022#discussion_r1597211056  

See this discussion

File: test/verify/test_conv_add_tune.cpp [hunk 5]
```diff
@@ -0,0 +1,76 @@
...
+ */
+
+#include "verify_program.hpp"
+#include <migraphx/program.hpp>
+#include <migraphx/generate.hpp>
+#include <migraphx/make_op.hpp>
+#include <migraphx/op/common.hpp>
+
+template <migraphx::shape::type_t DType>
+struct test_conv_add_tune : verify_program<test_conv_add_tune<DType>>
```
Comment (umangyadav) [inline test/verify/test_conv_add_tune.cpp:32]: @pfultz2  added test. 
Sizes seem large but i had to pick sizes for the convolution that would offload to MLIR by default and would pick split-k perf config as the fastest one. 

This test uses Split-k perf config for the convolution on MI100 . 

Without fix from this PR, this test fails.

**Review by umangyadav — APPROVED** [review]

File: test/verify/test_conv_add_tune.cpp (hunk 5 above)
Comment (umangyadav) [reply test/verify/test_conv_add_tune.cpp:32]: It is not guranteed to pick split-k perf config all the time though. So this test may be dud for those cases. On MI250, i do not see it picking split-k config as the fastest one.

**Review by causten — APPROVED** [review]

**Pull request merged**